### ISSUE TITLE
Feature request: anchor link for page title

### ISSUE THREAD

**Issue by jamesbraza** [thegeeklab/hugo-geekdoc#636]
From this page: https://geekdocs.de/usage/getting-started/

On [Install requirements](https://geekdocs.de/usage/getting-started/#install-requirements), there's an anchor link:

<img width="308" alt="image" src="https://github.com/thegeeklab/hugo-geekdoc/assets/8990777/c0b0f3d8-5a1c-4093-851e-0a83f43c0747">

On the Getting Started page's title, there is no anchor link:

<img width="286" alt="image" src="https://github.com/thegeeklab/hugo-geekdoc/assets/8990777/01d876fb-1d33-467b-9b15-54bf83912232">

Thoughts on supporting anchoring on the page title?

See `pylint`'s docs for a documentation website that supports what's being proposed: https://pylint.readthedocs.io/en/stable

**Comment by xoxys** [comment]
That's right, for the `h1` layer, we don't add anchors. The first level heading is at the top of the page anyway. Can you explain your use case? How does an anchor help with content that is already in the default view area?

**Comment by jamesbraza** [comment]
Oh the motivation is primarily symmetry/consistency of all headers in a page.  It just feels like there should be an anchor link for the title, when one has `geekdocAnchor = true`.

An anchor on the page title helps if there's an intro section before the first header.  The anchor link to the title would take one to just above the intro.

If you don't like it, then perhaps the default would be `false` for a new config option like `geekdocTitleAnchors`

**Comment by xoxys** [comment]
Was looking into it, but the problem is how the page title is handled and how we implemented anchors. 

To add anchors to headings, we use a feature called [markdown render hooks](https://gohugo.io/templates/render-hooks/). These hooks can be used to influence how some markdown elements are rendered, for the headings we inject HTML to add the anchors, see https://github.com/thegeeklab/hugo-geekdoc/blob/main/layouts/_default/_markup/render-heading.html

The page title `<h1>` is generated from the front matter `title` parameter and not processed by the markdown renderer of Hugo. As a result, the render hooks responsible for the anchors is bypassed. As the `title` element of a page is used in multiple places (same as discussed in the issue related to code blocks in the title) it would add a lot of cluttering to support anchors there. Hope that's understandable.
